src/backend/x11/mod.rs

```diff
@@ -441,7 +441,7 @@ impl X11Surface {
     /// This will fail if the backend has already been used to create a surface.
     pub fn new(
         backend: &mut X11Backend,
-        device: gbm::Device<DrmNode>,
+        device: Arc<Mutex<gbm::Device<DrmNode>>>,
```

Comment:
I wonder, if this is something we should rather handle internally.
Although you can now share allocators between different backends with this PR, users should probably not be encouraged to do that and rather create multiple `gbm::Device`s for different backends.

Reply:
Well the constructor for `X11Surface` needs to get the gbm device somehow. We don't make the X11Backend or X11Handle create the device at the moment.

To propagate the device to new X11Surfaces, we would need to either bootstrap off one surface or make an intermediate object for the device that internally holds the Mutex?

Reply:
I though about giving the `X11Backend` a `create_surface` function and making this constructor private.

Reply:
Hmm that seems like a fine way around this issue